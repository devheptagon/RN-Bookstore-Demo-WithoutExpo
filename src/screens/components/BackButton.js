import React from "react";
import PropTypes from "prop-types";
import Icon from "react-native-vector-icons/FontAwesome";
import { Text, StyleSheet, TouchableHighlight } from "react-native";
import FlexView from "./FlexView";
import Theme from "../../constants/Theme";
import { FlexDirections, IconNames } from "../../constants/Enums";
import Texts from "../../constants/Texts";

const BackButton = props => (
  <TouchableHighlight onPress={() => props.navigator.goBack(null)}>
    <FlexView flex={0} flexDirection={FlexDirections.Row} centerAll>
      <Icon name={IconNames.ChevronLeft} size={20} style={styles.Arrow} />
      <Text style={styles.TitleText}> {Texts.Back} </Text>
    </FlexView>
  </TouchableHighlight>
);

const styles = StyleSheet.create({
  TitleText: {
    color: Theme.HeaderBar.TextColor,
    fontSize: Theme.General.Text.FontXBig,
    fontWeight: Theme.General.Text.FontBold
  },
  Arrow: {
    marginRight: 5,
    marginTop: 5,
    color: Theme.HeaderBar.TextColor
  }
});

BackButton.propTypes = {
  navigator: PropTypes.object
};

export default BackButton;
