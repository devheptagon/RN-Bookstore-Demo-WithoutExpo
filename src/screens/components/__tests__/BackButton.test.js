import React from 'react';
import BackButton from '../BackButton';
import renderer from 'react-test-renderer';

test('BackButton renders correctly', () => {
  const tree = renderer.create(<BackButton />).toJSON();
  expect(tree).toMatchSnapshot();
});