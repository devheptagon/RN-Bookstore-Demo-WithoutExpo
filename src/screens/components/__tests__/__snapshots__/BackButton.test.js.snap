// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
       
      Back
       
    </Text>
  </View>
</View>
`;
